omegatrade-app/backend/app/models/company.model.js

```diff
@@ -0,0 +1,61 @@
+'user strict';
+const database = require('./../config/database.js');
+var Company = function () { };
+
+Company.getAll = async function (cb) {
+    try {
+        const query = { sql: 'select companyId , companyName , companyShortCode , created_at from companies', json: true, };
+        const [companies] = await database.run(query);
```

Comment:
Depending on the (expected) size of the results of the query, you should consider using `database.runStream(query)` instead of `database.run(query)`. `database.run(..)` will load the entire query result into memory, while `database.runStream(..)` will return a stream of rows. When you know that the result set will be relatively small (as in at most a couple of thousand rows), then this is fine.

For the table `companies` this is probably fine, as the application will probably never contain more than a few thousand companies, but for other tables you should either use `runStream` or add a `LIMIT`  clause to the query to be sure.

Reply:
yes, you are correct, here the companies tables are expected few thousand rows, so we didn't use pagination or limit in the backend. instead, we handled search,sort,paging features in the frontend.